framework/smoke_test_framework.cpp

```diff
@@ -0,0 +1,592 @@
+
+    add(&target_hostname,
+        "target_hostname",
+        "Target Slave. (Waiting for offer from specified slave.");
+
+  }
+
+  Option<string> master;
+  bool checkpoint;
+  Option<string> principal;
+  Option<string> secret;
+  string command;
+  string task_resources;
+  Option<string> task_revocable_resources;
+  Option<size_t> num_tasks;
+  Option<string> tasks_json_path;
+  Option<string> target_hostname;
+  string role;
+};
+
+
+int parseTaskJson(list<JobSpec>& jobs, const Flags flags, bool& revocable) {
+
+  LOG(INFO) << "Loading JSON with tasks from: " << flags.tasks_json_path.get();
+
+  Try<std::string> read = os::read(flags.tasks_json_path.get());
+  if (read.isError()) {
+    EXIT(EXIT_FAILURE) << read.error() << " "
+    << flags.usage("Bad path for JSON tasks");
+  } else if (read.get().empty()) {
+    EXIT(EXIT_FAILURE) << "File is empty.  "
+    << flags.usage("Bad path for JSON tasks");
+  }
+
+  Try<JSON::Object> json = JSON::parse<JSON::Object>(read.get());
+  if (json.isError()) {
+    EXIT(EXIT_FAILURE) << json.error() << " "
+    << flags.usage("Bad JSON file.");
+  }
+
+  Result<JSON::Array> tasks = json.get().find<JSON::Array>("tasks");
+  if (!tasks.isSome()) {
+    EXIT(EXIT_FAILURE)
+    << flags.usage("JSON file does not contain array 'tasks' in root.");
+  }
+
+  int numJobs = tasks.get().values.size();
+
+  for(int i=0; i<numJobs;i++) {
+    Result<JSON::String> _command =
+        json.get().find<JSON::String>("tasks[" + stringify(i) +"].command");
+
+    if (!_command.isSome()) {
+      EXIT(EXIT_FAILURE)
+      << flags.usage(
+          "JSON task " + stringify(i) + "does not contain command");
+    }
+
+    Result<JSON::String> _taskResources =
+        json.get().find<JSON::String>(
+            "tasks[" + stringify(i) +"].taskResources");
+
+    if (!_taskResources.isSome()) {
+      EXIT(EXIT_FAILURE)
+      << flags.usage(
+          "JSON task " + stringify(i) + "does not contain taskResources");
+    }
+
+    Try<Resources> _resources =
+        Resources::parse(_taskResources.get().value);
+
+    if (_resources.isError()) {
+      EXIT(EXIT_FAILURE)
+      << flags.usage("Invalid taskResources in JSON:" +
+                     _resources.error());
+    }
+
+    Result<JSON::String> _revocableTaskResources =
+        json.get().find<JSON::String>(
+            "tasks[" + stringify(i) +"].revocableResources");
+
+    if (_revocableTaskResources.isSome()) {
+      revocable = true;
+      Try<Resources> _revocableResources =
+          Resources::parse(_revocableTaskResources.get().value);
+
+      if (_revocableResources.isError()) {
+        EXIT(EXIT_FAILURE)
+        << flags.usage("Invalid revocableResources in JSON: " +
+                       _revocableResources.error());
+      }
+
+          foreach (Resource _revocable, _revocableResources.get()) {
+              _revocable.mutable_revocable();
+              _resources.get() += _revocable;
+            }
```

Comment:
No, since it's a mesos style file. I would have to  repair over 50 issues. And if we would like to upstream this i would have to revert this changes. ;p I don't know if we want to upstream.. but there is such possibility (: It would be useful.